src/player/update/update_hurt.c

```diff
@@ -5,22 +5,79 @@
 ** update hurt
 */
 
+#include <SFML/Graphics/View.h>
+#include <SFML/System/Vector2.h>
+#include <stdbool.h>
+#include "ennemies.h"
+#include "my_dico.h"
 #include "my_rpg.h"
 #include "my_json.h"
+#include "math.h"
+#include "maths.h"
 
-static void check_blink_time_end(float *since_start, float dtime, any_t *data,
+static const float time_nulll = 1;
+
+static const float coef = 1000;
+
+const char data_directions[] = "dirrections player hurt";
+
+static int nb_intervals = 20;
+
+static void update_move_player(player_t *player, window_t *win, sfVector2f new)
+{
+    ennemy_t enn = {0};
+    sfVector2f delta_x = {new.x / nb_intervals, new.y / nb_intervals};
+
+    enn = (ennemy_t) {player->state, player->dir, 1, player->obj, 0, 1};
+    for (int i = 0; i < nb_intervals; i++) {
+        if (check_wall(&enn, new, win) == true) {
+            break;
+        }
+        player->obj->bigdata.sprite_bigdata.pos.x += delta_x.x;
+        player->obj->bigdata.sprite_bigdata.pos.y += delta_x.x;
+        increment_hud_pos(win, get_distance(
+            player->obj->bigdata.sprite_bigdata.pos,
+            sfView_getCenter(player->view)));
+        sfView_setCenter(player->view, player->obj->bigdata.sprite_bigdata.pos);
+    }
+}
+
+static void update_player_rollback(player_t *player, float since_start,
+    window_t *win)
+{
+    sfVector2f *dirrection = NULL;
+    sfVector2f new = {0};
+
+    if (player == NULL || player->obj == NULL) {
+        return;
+    }
+    dirrection = dico_t_get_value(player->obj->components, data_directions);
+    if (dirrection == NULL) {
+        return;
+    }
+    new = (sfVector2f) {dirrection->x / since_start,
+        dirrection->y / since_start};
+    update_move_player(player, win, new);
+}
+
+static void check_blink_time_end(float *since_start, float dtime, window_t *win,
     player_t *player)
 {
-    any_t *blink_time = dico_t_get_any(data->value.dict, "blink time");
+    any_t *data = dico_t_get_value(player->obj->components, PLAYER_DATA);
+    any_t *blink_time = get_from_any(data, "d", "blink time");
 
     if (blink_time == NULL || blink_time->type != FLOAT || player == NULL ||
             player->obj == NULL) {
         return;
```

Comment:
Crazy ? Why you recheck the player whereas you re using it before and you check it in the main function ?? I think this check is useless because the function is static